string sentence
int32 count
---
bool finish